test/etl_test.py

```diff
@@ -131,3 +128,103 @@ def test_ferc1_solo_etl(datastore_fixture,
         pudl_settings_fixture,
         datapkg_bundle_name='ferc1-solo',
         clobber=True)
+
+
+class TestFerc1Datastore:
+    """Validate the Ferc1 Datastore and integration functions."""
+
+    def test_ferc_folder(self, pudl_ferc1datastore_fixture):
+        """Spot check we get correct folder names per dataset year."""
+        ds = pudl_ferc1datastore_fixture
+
+        assert ds.get_folder(1994) == "FORMSADMIN/FORM1/working"
+        assert ds.get_folder(2001) == "UPLOADERS/FORM1/working"
+        assert ds.get_folder(2002) == "FORMSADMIN/FORM1/working"
+        assert ds.get_folder(2010) == "UPLOADERS/FORM1/working"
+        assert ds.get_folder(2015) == "UPLOADERS/FORM1/working"
+
+    def test_get_fields(self, pudl_ferc1datastore_fixture):
+        """Check that the get fields table works as expected."""
+        ds = pudl_ferc1datastore_fixture
+
+        expect_path = pathlib.Path(__file__).parent / \
+            "data/ferc1/f1_2018/get_fields.json"
+
+        with expect_path.open() as f:
+            expect = yaml.safe_load(f)
+
+        data = ds.get_file(2018, "F1_PUB.DBC")
+        result = get_fields(data)
+        assert result == expect
+
+    def test_sample_get_dbc_map(self, pudl_ferc1datastore_fixture):
+        """Test sample_get_dbc_map."""
+        ds = pudl_ferc1datastore_fixture
+
+        table = get_dbc_map(ds, 2018)
+        assert table["f1_429_trans_aff"] == {
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/zaneselvans/pudl/pullRequest?prid=6263807)